gprofiler/merge.py

```diff
@@ -245,19 +246,20 @@ def concatenate_profiles(
     lines = []
 
     for pid, stacks in process_profiles.items():
-        container_name = _get_container_name(pid, docker_client)
+        container_name = _get_container_name(pid, docker_client, add_container_names)
         for stack, count in stacks.items():
             total_samples += count
-            lines.append(f"{container_name};{stack} {count}")
+            lines.append(f"{container_name + ';' if add_container_names else ''}{stack} {count}")
 
-    lines.insert(0, _make_profile_metadata(docker_client))
+    lines.insert(0, _make_profile_metadata(docker_client, add_container_names))
     return "\n".join(lines), total_samples
 
 
 def merge_profiles(
     perf_pid_to_stacks_counter: ProcessToStackSampleCounters,
     process_profiles: ProcessToStackSampleCounters,
     docker_client: Optional[DockerClient],
+    add_container_names: bool,
```

Comment:
Perhaps the API can be reworked in a way that `merge_profiles` doesn't have to receive `docker_client` & `add_container_names` as arguments?

Reply:
The problem is that it's two different situations - 
If the `docker_client` is None I still want to add a `;` at the start of each line, but not if the `add_container_names` is false. I don't think it's too bad sending these two different args

Reply:
Aha, because in `v2` we have to send it in any way. Oh well. This code will have to be restructured some when